app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -619,7 +621,7 @@ protected void onDestroy() {
     @Override
     public void onItemClick(AdapterView<?> listView, View view, int position, long id) {
         if (id == EntityListAdapter.SPECIAL_ACTION) {
-            triggerDetailAction();
+            triggerDetailAction(position - (adapter.getCount() - 1));

```

Comment:
It's a little bit obfuscated what these counts represent / what you're trying to calculate here; maybe worth adding a comment.